exonum/src/api/node/public/explorer.rs

```diff
@@ -246,24 +250,51 @@ impl ExplorerApi {
                 ApiError::NotFound(description)
             })
     }
+
     /// Adds transaction into unconfirmed tx pool, and broadcast transaction to other nodes.
     pub fn add_transaction(
-        state: &ServiceApiState,
-        query: TransactionHex,
-    ) -> Result<TransactionResponse, ApiError> {
-        use crate::events::error::into_failure;
-        use crate::messages::ProtocolMessage;
-
-        let buf: Vec<u8> = ::hex::decode(query.tx_body).map_err(into_failure)?;
-        let signed = SignedMessage::from_raw_buffer(buf)?;
-        let tx_hash = signed.hash();
-        let signed = RawTransaction::try_from(Message::deserialize(signed)?)
-            .map_err(|_| format_err!("Couldn't deserialize transaction message."))?;
-        let _ = state
-            .sender()
-            .broadcast_transaction(signed)
-            .map_err(ApiError::from);
-        Ok(TransactionResponse { tx_hash })
+        name: &'static str,
+        backend: &mut actix_backend::ApiBuilder,
+        service_api_state: ServiceApiState,
+    ) {
+        let snapshot = service_api_state.blockchain().snapshot();
+        let schema = Schema::new(&snapshot);
+        let max_message_len = schema.actual_configuration().consensus.max_message_len;
+
+        let handler = |state: &ServiceApiState,
+                       query: TransactionHex|
+         -> Result<TransactionResponse, ApiError> {
+            let buf: Vec<u8> = ::hex::decode(query.tx_body).map_err(into_failure)?;
+            let signed = SignedMessage::from_raw_buffer(buf)?;
+            let tx_hash = signed.hash();
+            let signed = RawTransaction::try_from(Message::deserialize(signed)?)
+                .map_err(|_| format_err!("Couldn't deserialize transaction message."))?;
+            let _ = state
+                .sender()
+                .broadcast_transaction(signed)
+                .map_err(ApiError::from);
+            Ok(TransactionResponse { tx_hash })
+        };
+
+        let index = move |request: HttpRequest| {
+            let state = request.state().clone();
```

Comment:
Well, it would allow to send transactions without 2x overhead; I wouldn't call that redundant 🙃 We probably should experiment a little, but it seems that sending binary data is quite feasible at least in the case of the JS client [via `fetch`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch):

```javascript
const body = new Uint8Array(32); // serialize message to binary...
fetch('/api/explorer/v1/transactions', {
    method: 'POST',
    headers: { 'Content-Type': 'application/octet-stream' },
    body,
})
```

Reply:
What about to leave this PR as a hot fix. In a PR with fresh `actix` we will add a possibility to send transactions as a binary data. Ok ?